### ISSUE TITLE
[YSQL] reltuples catalog table entries are not picked up for table estimates across sessions

### ISSUE THREAD

**Issue by tanujnay112** [yugabyte/yugabyte-db#18024]
Jira Link: [DB-7080](https://yugabyte.atlassian.net/browse/DB-7080)
### Description

It would appear that the result of `ANALYZE` on system tables are persisted in `reltuples` of `pg_class` across sessions but query plans are only able to use these for row estimates in the session where `ANALYZE` was run.

```
yugabyte=# explain select * from pg_attribute;
                            QUERY PLAN                             
-------------------------------------------------------------------
 Seq Scan on pg_attribute  (cost=0.00..100.00 rows=1000 width=237)
(1 row)

yugabyte=# analyze;
WARNING:  'analyze' is a beta feature!
LINE 1: analyze;
        ^
HINT:  Set 'ysql_beta_features' yb-tserver gflag to true to suppress the warning for all beta features.
ANALYZE
yugabyte=# explain select * from pg_attribute;
                            QUERY PLAN                             
-------------------------------------------------------------------
 Seq Scan on pg_attribute  (cost=0.00..252.00 rows=2520 width=237)
(1 row)

yugabyte=# ^D\q
tn-mbp-vmd6n:yugabyte-db tnayak$ ./bin/ysqlsh
ysqlsh (11.2-YB-2.19.1.0-b0)
Type "help" for help.

yugabyte=# 
yugabyte=# explain select * from pg_attribute;
                            QUERY PLAN                             
-------------------------------------------------------------------
 Seq Scan on pg_attribute  (cost=0.00..100.00 rows=1000 width=237)
(1 row)
```
```
yugabyte=# explain select * from pg_attribute;
                            QUERY PLAN                             
-------------------------------------------------------------------
 Seq Scan on pg_attribute  (cost=0.00..100.00 rows=1000 width=237)
(1 row)

Time: 31.734 ms
yugabyte=# select reltuples from pg_class where relname = 'pg_attribute';
 reltuples 
-----------
      2520
(1 row)

Time: 5.620 ms 
```

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-7080]: https://yugabyte.atlassian.net/browse/DB-7080?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by tanujnay112** [comment]
It looks like this is a result of reading stale pg_class cached entries. The problem goes away when a cache invalidation is added via `RelationCacheInvalidateEntry(relationObjectId);` over [here](https://github.com/yugabyte/yugabyte-db/blob/1de3265290edd614dbbb25d00fec78068f269c3f/src/postgres/src/backend/optimizer/util/plancat.c#L121).

**Comment by yifanguan** [comment]
Resolved by commit 9a27affa150e708573a815a0e1ed9edd25cf6424
